python/demo/poisson/demo_poisson.py

```diff
@@ -134,10 +141,19 @@
 # immediately above. The definition of the Dirichlet boundary condition
 # then looks as follows: ::
 
+# Create a function that returns True if the point is on the boundary
+
+
+def marker(x):
+    left = np.isclose(x[0], 0.0)
+    right = np.isclose(x[0], 2.0)
+    return np.logical_or(left, right)
+
+
 # Define boundary condition on x = 0 or x = 1
-facets = locate_entities_boundary(mesh, 1, lambda x: np.logical_or(np.isclose(x[0], 0.0),
-                                                                   np.isclose(x[0], 2.0)))
-bc = dirichletbc(ScalarType(0), locate_dofs_topological(V, 1, facets), V)
+facets = _mesh.locate_entities_boundary(mesh, dim=1, marker=marker)
+dofs = fem.locate_dofs_topological(V=V, entity_dim=1, entities=facets)
+bc = fem.dirichletbc(value=ScalarType(0), dofs=dofs, V=V)
```

Comment:
`ScalarType(0)` is clumsy and really just for the CI. Would be good if we could avoid it somehow, without reducing the functionality.